torch/csrc/autograd/profiler_kineto.h

```diff
@@ -0,0 +1,212 @@
+#pragma once
+
+#include <torch/csrc/autograd/profiler_legacy.h>
+
+#include <c10/core/DeviceType.h>
+//#include <torch/csrc/WindowsTorchApiMacro.h>
+
+#ifdef USE_KINETO
+namespace libkineto {
+class TraceActivity;
+class ActivityTraceInterface;
+}
+#endif
+
+namespace torch {
+namespace autograd {
+namespace profiler {
+
+enum class C10_API_ENUM ActivityType {
+  CPU = 0,
+  // CUDA_RUNTIME, // CUDA host events
+  CUDA, // CUDA kernels
+  NUM_KINETO_ACTIVITIES, // must be the last one
+};
+
+#ifdef USE_KINETO
+
+struct KinetoObserverContext : public at::ObserverContext {
+  int64_t startUs;
+  uint64_t correlationId;
+  uint64_t startThreadId;
+  uint64_t endThreadId;
+  c10::optional<std::vector<std::vector<int64_t>>> shapes;
+  int64_t sequenceNr;
+  uint64_t fwdThreadId;
+  uint8_t recFunScope;
+  c10::optional<std::vector<std::string>> stack;
+};
+
+struct TORCH_API KinetoEvent {
```

Comment:
Seems like we should be able to do better here and clean this up, e.g. make KinetoEvent inherit from ClientTraceActivity. We can discuss follow-up.